plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -143,6 +143,13 @@ public void testSortItemsReflectedInExplain()
                 "TopNPartial\\[count = 5, orderBy = \\[nationkey DESC");
     }
 
+    @Test
+    public void testProjectionPushdown()
```

Comment:
This test would pass even without `applyProjection` being implemented right ?

Reply:
Yes, right. I failed to understand where simple projection will get benefited.

Reply:
I don't think this would pass without `applyProjection` since a `ProjectNode` will exist above the `TableScanNode`.

However once deref pushdown is also implemented we can change the test to verify that table scan read less data compared to SELECT * query.